File: app/src/main/java/com/diogenes/userswitcher/MainActivity.java
package com.diogenes.userswitcher;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

public class MainActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        try {
            Intent intent = new Intent("android.settings.USER_SETTINGS");
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
            startActivity(intent);
        } catch (Exception e) {
            Toast.makeText(this, "Ошибка запуска меню", Toast.LENGTH_SHORT).show();
        }
        finish();
    }
}

File: app/src/main/java/com/android/userswitcher/MainActivity.kt
package com.android.userswitcher

import android.content.ActivityNotFoundException
import android.content.Intent
import android.os.Bundle
import android.provider.Settings
import androidx.activity.ComponentActivity

class MainActivity : ComponentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        try {

            startActivity(
                Intent(Settings.ACTION_USER_SETTINGS)
            )

        } catch (_: ActivityNotFoundException) {

            startActivity(
                Intent(Settings.ACTION_SETTINGS)
            )

        }

        finish()
    }
}